bcda/database/utils.go

```diff
@@ -3,12 +3,16 @@ package database
 import (
 	"runtime"
 
-	"github.com/jinzhu/gorm"
 	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
 )
 
 func Close(db *gorm.DB) {
-	if err := db.Close(); err != nil {
+	dbc, err := db.DB()
+	if err != nil {
+		log.Infof("failed to retrieve db connection: %v", err)
```

Comment:
We should do an early return if we fail to get the connection. Otherwise, `dbc` may be nil and we'll panic.